nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-configuration-history/ui/flow-configuration-history-listing/flow-configuration-history-table/flow-configuration-history-table.component.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MatTableDataSource, MatTableModule } from '@angular/material/table';
+import { MatSortModule, Sort } from '@angular/material/sort';
+import { ActionEntity } from '../../../state/flow-configuration-history-listing';
+import { NiFiCommon } from '../../../../../service/nifi-common.service';
+
+@Component({
+    selector: 'flow-configuration-history-table',
+    standalone: true,
+    imports: [CommonModule, MatTableModule, MatSortModule],
+    templateUrl: './flow-configuration-history-table.component.html',
+    styleUrls: ['./flow-configuration-history-table.component.scss']
+})
+export class FlowConfigurationHistoryTable {
+    @Input() selectedHistoryActionId: number | null = null;
```

Comment:
Does this need to be an `@Input` and saved in the state. We have a few other cases of table selection scenarios where we are not deep linking and the component just internally manages the selected row.

Reply:
It doesn't "need" to be. But it was a simple enough change to migrate from using the selection from the route to this. And it sets us up to change back to use the route for selection in the future if we choose.